### PR TITLE
✨ Add / for ScalarQuadraticFunction

### LINKED ISSUE #672: Missing / for quadratic functions
### What is the issue

The method `operate(::typeof(/), ::Type{T}, ::MOI.ScalarQuadraticFunction{T}, ::T)` is missing.

```julia
julia> using MathOptInterface

julia> const MOI = MathOptInterface
MathOptInterface

julia> v = MOI.SingleVariable(MOI.VariableIndex(1))
MathOptInterface.SingleVariable(MathOptInterface.VariableIndex(1))

julia> quad = 2.0v * v
MathOptInterface.ScalarQuadraticFunction{Float64}(MathOptInterface.ScalarAffineTerm{Float64}[ScalarAffineTerm{Float64}(0.0, VariableIndex(1))], MathOptInterface.ScalarQuadraticTerm{Float64}[ScalarQuadraticTerm{Float64}(2.0, VariableIndex(1), VariableIndex(1))], 0.0)

julia> quad / 2.0
ERROR: MethodError: no method matching operate(::typeof(/), ::Type{Float64}, ::MathOptInterface.ScalarQuadraticFunction{Float64}, ::Float64)
Closest candidates are:
  operate(::typeof(+), ::Type{T}, ::Any, ::Any, ::Any, ::Any...) where T at /home/blegat/.julia/dev/MathOptInterface/src/Utilities/functions.jl:799
  operate(::typeof(/), ::Type{T}, ::MathOptInterface.SingleVariable, ::T) where T at /home/blegat/.julia/dev/MathOptInterface/src/Utilities/functions.jl:1227
  operate(::typeof(/), ::Type{T}, ::MathOptInterface.ScalarAffineFunction{T}, ::T) where T at /home/blegat/.julia/dev/MathOptInterface/src/Utilities/functions.jl:1240
  ...
Stacktrace:
 [1] /(::MathOptInterface.ScalarQuadraticFunction{Float64}, ::Float64) at /home/blegat/.julia/dev/MathOptInterface/src/Utilities/functions.jl:1247
 [2] top-level scope at none:0
```

### Steps to solve the issue

1. Implement the method getting inspiration from
 https://github.com/JuliaOpt/MathOptInterface.jl/blob/1bdae588867d2c6a88485c94cba9e29ca6e76fbb/src/Utilities/functions.jl#L1232-L1241
2. Add tests in `test/Utilities/functions.jl`

### PR DESCRIPTION
**Pull request by Peiffap**
Implements the division operation for ScalarQuadraticFunction.
I still need to add tests but had a hard time finding where to add them...

- [x] Needs tests

Closes #672.

### CODE REVIEW

**Comment by blegat** [comment]
You can add tests between these two lines:
https://github.com/JuliaOpt/MathOptInterface.jl/blob/1bdae588867d2c6a88485c94cba9e29ca6e76fbb/test/Utilities/functions.jl#L400-L401

File: src/Utilities/functions.jl [hunk 1]
```diff
@@ -1240,6 +1240,18 @@ function operate(::typeof(/), ::Type{T}, f::MOI.ScalarAffineFunction{T},
...
 end
 
+function operate!(::typeof(/), ::Type{T}, f::MOI.ScalarQuadraticFunction{T},
+                  α::T) where T
+    f.aff_terms .= operate_term.(/, f.affine_terms, α)
+    f.quad_terms .= operate_term.(/, f.quadratic_terms, α)
+    f.constant /= α
+    return f
+end
+function operate(::typeof(/), ::Type{T}, f::MOI.ScalarQuadraticFunction{T},
```
Comment (blegat) [inline src/Utilities/functions.jl]: We can merge this method with the `ScalarAffineFunction` one by doing `f::Union{MOI.ScalarAffineFunction{T}, MOI.ScalarQuadraticFunction{T}}`

**Comment by Peiffap** [comment]
I'm gonna look into refactoring the code a bit more, just to keep it as short and simple as possible.

File: src/Utilities/functions.jl [hunk 2]
```diff
@@ -1235,8 +1235,17 @@ function operate!(::typeof(/), ::Type{T}, f::MOI.ScalarAffineFunction{T},
...
+function operate!(::typeof(/), ::Type{T}, f::MOI.ScalarQuadraticFunction{T},
+                  α::T) where T
+    f.affine_terms .= operate_term.(/, f.affine_terms, α)
+    f.quadratic_terms .= operate_term.(/, f.quadratic_terms, α)
+    f.constant /= α
+    return f
+end
+
+function operate(::typeof(/), ::Type{T},
+                 f::Union{MOI.ScalarAffineFunction{T},                                 MOI.ScalarQuadraticFunction{T}}, α::T) where T
```
Comment (blegat) [inline src/Utilities/functions.jl]: Missing a newline here it seems

Comment (Peiffap) [reply src/Utilities/functions.jl]: Oh, automatic line-wrapping confused me again 😭

Comment (Peiffap) [inline src/Utilities/functions.jl]:
```suggestion
                 f::Union{MOI.ScalarAffineFunction{T},
                          MOI.ScalarQuadraticFunction{T}}, α::T) where T
```

File: test/Utilities/functions.jl [hunk 3]
```diff
@@ -397,7 +397,10 @@ const MOIU = MOI.Utilities
                     MOI.SingleVariable(x)
                 end
                 @test f ≈ MOI.ScalarQuadraticFunction(MOI.ScalarAffineTerm.([3], [x]),
-                                                      MOI.ScalarQuadraticTerm.([1, 2, 3], [x, y, x], [x, y, y]), 10) - 3
+                                              MOI.ScalarQuadraticTerm.([1, 2, 3], [x, y, x], [x, y, y]), 10) - 3
+                @test f ≈
```
Comment (blegat) [inline test/Utilities/functions.jl]: Why not simply `@test f ≈ 2.0 * f / 2.0` ?

Comment (Peiffap) [reply test/Utilities/functions.jl]: I honestly am not familiar with Julia enough, I was scared that might be optimised out by the compiler at run time. I also looked at the other tests, saw you didn't do `@test f ≈  f + 3 - 3`, and figured that must've been for a reason.

Comment (blegat) [reply test/Utilities/functions.jl]: Assuming that multiplication works, `f ≈ 2.0 * f / 2.0` correctly tests division. For the starting tests, you cannot assume that any operation works so you need to start with the more verbose style, e.g. `f + 3 ≈ MOI.ScalarAffineFunction(...)`. Then `f ≈ (f + 3) - 3` should be a decent test for `-`

Comment (Peiffap) [reply test/Utilities/functions.jl]: That makes sense, I'll change it.

Comment (Peiffap) [reply test/Utilities/functions.jl]: Oh, I just realised why I didn't do that. `f` is defined as a `ScalarQuadraticFunction{Int64}`, while `2.0` is a `Float64`.

Comment (blegat) [reply test/Utilities/functions.jl]: Indeed, it works because `isapprox` works for functions of different coefficient type `T`

**Comment by blegat** [comment]
Thanks for the PR ! :)

**Pull request merged**